### PR TITLE
Cli: list-invitations

### LINKED ISSUE #6067: CLI `list-invitations`

### PR DESCRIPTION
**Pull request by TimeEngineer**
closes #6067

### CODE REVIEW

File: cli/src/list_invitations.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                        "Server error while listing invitations: {rep:?}"
+                    ));
+                }
+            };
+
+            handle.done();
+
+            if invitations.is_empty() {
+                println!("No invitation.");
+            }
```
Comment (FirelightFlagboy) [inline cli/src/list_invitations.rs:45]:
```suggestion
                println!("No invitation.");
                return Ok(());
            }
```

I would prefer to have an early return or have an `if .. else ..` block

File: cli/src/list_invitations.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                        (token, status, "device".into())
+                    }
+                };
+
+                let token = token.as_simple();
+
+                let display_status = match status {
+                    InvitationStatus::Idle => format!("{YELLOW}idle{RESET}"),
+                    InvitationStatus::Ready => format!("{GREEN}idle{RESET}"),
+                    InvitationStatus::Deleted => format!("{RED}idle{RESET}"),
```
Comment (FirelightFlagboy) [inline cli/src/list_invitations.rs:65]: Is it expected that all text is `idle` ?

File: cli/src/tests.rs [hunk 3]
```diff
@@ -253,3 +253,58 @@ async fn cancel_invitation(tmp_path: TmpPath) {
             "Deleting invitation\nInvitation deleted",
         ));
 }
+
+#[rstest::rstest]
+#[tokio::test]
+async fn list_invitations(tmp_path: TmpPath) {
```
Comment (FirelightFlagboy) [inline cli/src/tests.rs:259]: Is it possible to have the other cases tested (`Deleted` & `Ready`) ?

File: cli/src/list_invitations.rs (hunk 2 above)
Comment (TimeEngineer) [reply cli/src/list_invitations.rs:65]: No lol

File: cli/src/tests.rs (hunk 3 above)
Comment (TimeEngineer) [reply cli/src/tests.rs:272]: I don't know how to reproduce them

Comment (TimeEngineer) [reply cli/src/tests.rs:272]: If you delete the `invitation`, it will be absent instead of the status `deleted`

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**